clippy_lints/src/literal_representation.rs

```diff
@@ -2,7 +2,7 @@
 //! floating-point literal expressions.
 
 use rustc::lint::*;
-use rustc::{declare_lint, lint_array};
+use rustc::{declare_tool_lint, lint_array};
```

Comment:
Why do we even need this imported here? Shouldn't `declare_clippy_lint` be enough? Maybe we need to import it in the expansion of `declare_clippy_lint`

Reply:
Because we use the `declare_clippy_lint!` which expands to `...declare_(tool_)lint!...`. 

> Maybe we need to import it in the expansion of declare_clippy_lint

That would probably be cleaner.

Reply:
Or not: If we use `declare_clippy_lint` multiple times in one file we would get multiple `use rustc::declare_tool_lint`. And that happens pretty often.

Reply:
can we just call `rustc::declare_tool_lint!` instead of importing it?

Reply:
If we use `rustc::declare_tool_lint!` in the macro_rule, we have to import `use rustc` in each file, or else:
```bash
error[E0433]: failed to resolve. Use of undeclared type or module `rustc`
  --> clippy_lints/src/lib.rs:24:9
   |
24 |           rustc::declare_tool_lint! { pub clippy::$name, Warn, $description, report_in_external_macro: true }
   |           ^^^^^ Use of undeclared type or module `rustc`
   |
  ::: clippy_lints/src/bit_mask.rs:87:1
```

Reply:
:( ok, lets leave it as is then and punt to the future